src/lib/OpenEXRUtil/ImfCheckFile.cpp

```diff
@@ -521,9 +529,14 @@ readDeepTile(T& in,bool reduceMemory , bool reduceTime)
 
         DeepFrameBuffer frameBuffer;
 
-        int memOffset = dataWindow.min.x + dataWindow.min.y * width;
+        //
+        // use 64 bit integer arithmetic instead of pointer arithmetic to compute offset into array. Pointer arithmetic may overflow
+        // memOffset is difference in bytes between theoretical address of pixel (0,0) and the origin of the data window
+        //
+        Int64 memOffset = sizeof(unsigned int) * (static_cast<Int64>(dataWindow.min.x) + static_cast<Int64>(dataWindow.min.y) * width);
+        intptr_t base = reinterpret_cast<intptr_t>(&localSampleCount[0][0] );
         frameBuffer.insertSampleCountSlice (Slice (UINT,
-                                                   (char *) (&localSampleCount[0][0] - memOffset),
+                                                   reinterpret_cast<char*> (base - memOffset),
```

Comment:
Does base need to be an Int64 as well? On a 32 bit architecture, inptr_t will be 32 bits, and so the calculation base - memOffset is into the realm of undefined behavior. (Ditto the question everywhere intptr_t is used.)

Reply:
base-memOffset can indeed be negative, since it's computing a theoretical address. The use of intptr_t is really to sneak past the detection of exactly that undefined behavior. I'd understood that intptr_t is supposed to be used to do calculations on memory addresses with 'normal' integers, so thought it was the right choice. On 32 bit architectures, am I right in assuming that since `memOffset` is Int64, `base` will be cast to an Int64 for the subtraction anyway?

So far, the Imf::Slice API works, even though it requires negative addresses and also relies on 64 bit computation wrapping round correctly to give 32 bit addresses on 32 bit architectures.

Reply:
You're right, the subtraction will occur as int64, so it is fine.